### PR TITLE
fix(script): Update scripts used to run planner

### PR DESCRIPTION
**Pull request by jarifibrahim**
This PR fixes
https://openshift.io/openshiftio/openshiftio/plan/detail/2277
https://openshift.io/openshiftio/openshiftio/plan/detail/2278

**How to test -**
The runner scipt supports following flags
1. `-r` - This would reinstall everything
2. `-s` - This would start planner in standalone mode (using the runtime)
3. `-p` - Set path to planner directory (requires path argument)
4. `-f` - Set path to platform directory (requires path argument)

**Run the script as**
- `scripts/run-planner.sh -r -s` - if you wish to reinstall planner and run planner in standalone mode
- `scripts/run-planner.sh -r -f /home/User/fabric8-ui` - if you wish to set the platform path. By default, the platform path is assumed to me `Planner_Path/..`

**NOTE** - Integrated mode is the default mode for running planner via this script

If you start planner in 
- `Standalone Mode` - it should be accessible at `http://localhost:8008`
- `Integrated Mode` - It should be accessible at `http://localhost:3000`

**NOTE** - The `Standalone Mode` uses prod-preview backend by default. You can change this by setiing approriate environment variables (`FABRIC8_WIT_API_URL`, `FABRIC8_AUTH_API_URL`, etc)

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @jarifibrahim for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/b1cd45eb869dde3953f5907f8a3fb0c59c84e7fa/test-keeper.yml).

**Comment by michaelkleinhenz** [comment]
I still have issues there running it:

When running a vanilla branch with ```scripts/run-planner.sh --standalone --reinstall``` is crashes with "gulp: command not found". We should probably test if gulp is available and do an ```npm install``` first (or do it every time).

If I do ```npm install && scripts/run-planner.sh --standalone --reinstall```, it has an error on the ```rimraf``` step in ```dist``` (as the dist contents do not exist yet). After that build continues, but fails with:

```
> fabric8-planner@0.0.0-development build /tmp/fabric8-planner
> gulp build "--watch"

npm ERR! code EINVALIDTYPE
npm ERR! typeerror Error: Argument #2: Expected array but got string
npm ERR! typeerror     at EventEmitter.install (/home/kleinhenz/.nave/installed/8.3.0/lib/node_modules/npm/lib/install.js:184:3)
npm ERR! typeerror     at Object.commandCache.(anonymous function) (/home/kleinhenz/.nave/installed/8.3.0/lib/node_modules/npm/lib/npm.js:150:13)
npm ERR! typeerror     at /home/kleinhenz/.nave/installed/8.3.0/lib/node_modules/npm/lib/link.js:93:24
npm ERR! typeerror     at /home/kleinhenz/.nave/installed/8.3.0/lib/node_modules/npm/node_modules/graceful-fs/polyfills.js:284:29
npm ERR! typeerror     at FSReqWrap.oncomplete (fs.js:152:21)
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <https://github.com/npm/npm/issues>
```

and as a followup because the ```runtime/``` deps does not get fully installed:

```
sh: webpack-dev-server: Command not found.
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
```

The latter is a followup issue though, because the installation of the deps in ```runtime``` stops with the first error and webpack never gets installed.

**Comment by michaelkleinhenz** [comment]
The ```Error: Argument #2: Expected array but got string``` probably has something to do with the linking of ```../dist```.

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "build": "gulp build",
     "clean": "gulp clean",
+    "reinstall": "rm -rf node_modules && npm  --force cache clean && npm install",
```
Comment (debloper) [inline package.json]: You can already use `npm run clean -- --modules --cache && npm install`
New npm command works against the last script purge to keep it lean and efficient.

Using a makefile to bundle the script primitives into a full process to do something can be the next step.
Adding more and more scriptlet to npm-scripts isn't the best way to achieve that as it gets messy over time.

File: runtime/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "clean:dist": "npm run rimraf -- dist",
     "clean:install": "npm set progress=false && npm install",
     "clean": "npm --force cache clean && npm run rimraf -- node_modules doc coverage dist compiled dll",
-    "reinstall": "rimraf node_modules && npm --force cache clean && npm install",
+    "reinstall": "rm -rf node_modules && npm --force cache clean && npm install",
```
Comment (debloper) [inline runtime/package.json]: Finally someone did it.
Although, with this, people on Windows can't use the `reinstall` script anymore (is there anyone who uses Windows?).

File: package.json (hunk 1 above)
Comment (jarifibrahim) [reply package.json]: @debloper `npm run clean` uses `gulp` to clean/remove files. What if the user doesn't have globally installed `gulp` ?

File: runtime/package.json (hunk 2 above)
Comment (jarifibrahim) [reply runtime/package.json]: I don't think anyone uses windows. ;)

**Comment by michaelkleinhenz** [comment]
Ok, works for me now, only issue is that integrated does not pick up and recompile on changes to the Planner codebase. I am unsure if that was in the original script's integrated featureset, but I integrated is now the dev default, this is a required functionality.

**Comment by michaelkleinhenz** [comment]
Ok, works now!

**Comment by rgarg1** [comment]
@jarifibrahim Kindly update the docs along with the PR.

**Comment by jarifibrahim** [comment]
> @jarifibrahim Kindly update the docs along with the PR.

@rgarg1 Updating the docs is a seperate task (https://openshift.io/openshiftio/openshiftio/plan/detail/2280) and @Preeticp has volunteered to take it up. We can update the documentation as a seperate PR.

**Comment by jarifibrahim** [comment]
> @jarifibrahim Script is not getting kill on Ctrl+c. It keep running in background.

What command did you run?

**Comment by vikram-raj** [comment]
@jarifibrahim This one ```run-planner.sh -r -s```.

**Comment by jarifibrahim** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Pull request merged**